gtdynamics/optimizer/EqualityConstraint.cpp

```diff
@@ -16,7 +16,7 @@
 namespace gtdynamics {
 
 gtsam::NoiseModelFactor::shared_ptr DoubleExpressionEquality::createFactor(
-    const double mu, boost::optional<gtsam::Vector&> bias) const {
+    const double mu, std::optional<gtsam::Vector> bias) const {
```

Comment:
Notice the ref went away, wondering if there's anything to gain from using `std::ref` around vector (to avoid copies) or not.

Reply:
I was wondering about that too. `std::optional` doesn't like references right out of the box and GTSAM doesn't use reference types for optionals so figured this is okay.